spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/DefaultSessionFactoryLocator.java

```diff
@@ -54,7 +54,7 @@ public DefaultSessionFactoryLocator(Map<Object, SessionFactory<F>> factories, Se
 	 * @param key the lookup key.
 	 * @param factory the factory.
 	 */
-	public void addSessionFactory(String key, SessionFactory<F> factory) {
+	public void addSessionFactory(Object key, SessionFactory<F> factory) {
```

Comment:
How about this my concern: https://github.com/spring-projects/spring-integration/issues/3169#issuecomment-583058147 ?

Whenever we are going to back-port it we should keep backward compatibility, so an upgrade for the target project will be smooth. Even if we don't back-port it into a previous version, we still need to keep binary compatibility for current `5.3` version: Spring Cloud is going to support both Spring Boot 2.2 and 2.3 generations.

Therefore my concern about deprecating the current method and introduce a new one like you have done.

Thanks

Reply:
Done. Thanks for quick answer